applications/ShapeOptimizationApplication/python_scripts/custom_ios/wrl_reader.py

```diff
@@ -0,0 +1,98 @@
+# ==============================================================================
+#  KratosShapeOptimizationApplication
+#
+#  License:         BSD License
+#                   license: ShapeOptimizationApplication/license.txt
+#
+#  Main authors:    Geiser Armin, https://github.com/armingeiser
+#
+# ==============================================================================
+
+class Shape:
+    def __init__(self):
+        self.name = None
+        self.nodes = []
+        self.triangles = []
+
+
+def read_nodes(line, file):
+    nodes = []
+    while True:
+        if "[" in line:
+            break
+
+    line = next(file)
+
+    while True:
+        if "]" in line:
+            break
+        elif line == "\n":
+            line = next(file)
+
+        try:
+            entries = line.split(",")[0].split()
+            if len(entries) != 3:
+                raise RuntimeError("Did not find 3 coordinate components!", line)
+            nodes.append([float(x) for x in entries])
+        except IOError:
+            pass
+
+        line = next(file)
+
+    return nodes
+
+
+def read_triangles(line, file):
+    triangles = []
+    while True:
+        if "[" in line:
+            break
+
+    line = next(file)
+
+    while True:
+        if "]" in line:
+            break
+        elif line == "\n":
+            line = next(file)
+
+        try:
+            entries = line.split(",")[:3]
+            if len(entries) != 3:
+                raise RuntimeError("Did not find 3 triangle node indices!", line)
+            triangles.append([int(x) for x in entries])
+        except IOError:
+            pass
+
+        line = next(file)
+
+    return triangles
+
+
+def read_shape(line, file):
```

Comment:
I think its cleaner if you first create all variables and then define the initialize the shape with input variables. Rather than first initializing an empty shape and then define the member variables. Of course, this also implies a corresponding adjustment of the shape class.